sources/ClangSharp/Cursors/Decls/CXXRecordDecl.cs

```diff
@@ -0,0 +1,41 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using ClangSharp.Interop;
+
+namespace ClangSharp
+{
+    public class CXXRecordDecl : RecordDecl
+    {
+        private readonly Lazy<IReadOnlyList<CXXBaseSpecifier>> _bases;
+        private readonly Lazy<IReadOnlyList<CXXConstructorDecl>> _ctors;
+        private readonly Lazy<CXXDestructorDecl> _destructor;
+        private readonly Lazy<IReadOnlyList<FriendDecl>> _friends;
+        private readonly Lazy<IReadOnlyList<CXXMethodDecl>> _methods;
+        private readonly Lazy<IReadOnlyList<CXXBaseSpecifier>> _vbases;
+
+        internal CXXRecordDecl(CXCursor handle, CXCursorKind expectedKind) : base(handle, expectedKind)
+        {
+            _bases = new Lazy<IReadOnlyList<CXXBaseSpecifier>>(() => CursorChildren.Where((cursor) => cursor is CXXBaseSpecifier).Cast<CXXBaseSpecifier>().ToList());
```

Comment:
Most of the traversal logic was changed to be lazy and to just pull any specific exposures out via caching the result of a LINQ operation.